### PR TITLE
Creating BigQueryReadClient with UnboundedScheduledExecutorService

### PR DESCRIPTION
**Pull request by suryasoma**
Updating the default ExecutorProvider

### CODE REVIEW

File: bigquery-connector-common/pom.xml [hunk 1]
```diff
@@ -128,6 +128,11 @@
       <artifactId>gson</artifactId>
       <version>2.9.1</version>
     </dependency>
+    <dependency>
+      <groupId>org.apache.beam</groupId>
+      <artifactId>beam-sdks-java-core</artifactId>
+      <version>2.43.0</version>
```
Comment (davidrabinowitz) [inline bigquery-connector-common/pom.xml:134]: This version has [2 CVEs](https://mvnrepository.com/artifact/org.apache.beam/beam-sdks-java-core/2.43.0), latest (2.46.0) is ok.

Also, the jar is 5MB, do we need it all for just one class? As an alternative - perhaps we can trim the redundant classes using the filter option in the shade plugin

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: bigquery-connector-common/pom.xml [hunk 2]
```diff
@@ -128,6 +128,12 @@
       <artifactId>gson</artifactId>
       <version>2.9.1</version>
     </dependency>
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>annotations</artifactId>
+      <version>3.0.1</version>
+      <scope>provided</scope>
```
[the remark this answers is not in the document]
Comment (suryasoma) [reply bigquery-connector-common/pom.xml:135]: `spark-bigquery-parent` doesn't have ` <artifactId>annotations</artifactId>`

**Review by davidrabinowitz — APPROVED** [review]

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
Unit test is missing

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**